flows/contact.go

```diff
@@ -400,13 +400,13 @@ func (c *Contact) UpdatePreferredChannel(channel *Channel) bool {
 
 		for _, urn := range c.urns {
 			// tel URNs can be re-assigned, other URN schemes are considered channel specific
-			if urn.URN().Scheme() == urns.TelScheme && channel.SupportsScheme(urns.TelScheme) {
+			if urn.URN().Scheme() == urns.TelScheme && channel.SupportsScheme(urns.TelScheme) && channel.HasRole(assets.ChannelRoleSend) {
```

Comment:
I think this fix has it backward.. if we allow channel roles to be changed then we're accepting the URNs may be associated with a channel that can't send. 

I think you need to tweak `ChannelAssets.GetForURN` so it only uses the URN channel if it has the correct role.

Reply:
While channels being mutable does bring up the possibility that a previously preferred channel is no more, it does seem like we would want to avoid overwriting previous affinity when we know a channel isn't eligible to send no?

IE, I think we need both no?

Reply:
Right, I think we still want to prevent the channel to be set on the URL if we know we cannot send with that.
And the flow editor will also allow to use channel that can sent to be used to update a contact

I am adding the changes for `GetForURN`

Reply:
If that's the case (which I think does make sense) then the action should create an error event rather than just having this method silently do nothing. 

And possibly the editor shouldn't allow it in the first place but we can worry about that another time.